src/blog.tsx

```diff
@@ -1,59 +1,112 @@
-          </div>
-        </section>
-      </div>
-
-      <section className='ph3 measure-wide pv4 center'>
-        <ul className='list pl0'>
-          {getBlogPosts(website).map(it => {
-            return (
-              <li className='pv3'>
-                <h2 className='mb0'>
-                  <a href={getFullUrl(it)} className='link navy'>
-                    {it.payload.title}
-                  </a>
-                </h2>
-                <p className='black-40 mt1'>{moment(it.payload.publishedDate).format('MMMM Do YYYY')}</p>
-                <p className='lh-copy black-70'>{it.payload.description}</p>
-              </li>
-            )
-          })}
-        </ul>
-      </section>
-
-      <Consultation />
-      <Footer />
-    </Layout>,
-  )
-}
+import React from 'react'
+import moment = require('moment')
+import { Navbar, Html, Head, OpenGraph, Body, Footer, Consultation } from './layout.v3'
+import { join } from 'path'
+import { getOpenGraph, getPages, getConfig, State, Actions, Action, getBlogPosts, hasTag } from './store'
+import { defaultAssetsPipeline } from './optimise'
+import { Store } from 'redux'
+import { BlogPost } from './store/websiteReducer'
+import { VReference } from './files'
+
+export const Details = {
+  type: 'blog',
+  url: '/blog',
+  seoTitle: 'Blog ♦︎ Learnk8s',
+  title: 'Blog',
+  description: 'The fastest way to become an expert in deploying applications at scale with Kubernetes.',
+  openGraphImage: <img src='assets/open_graph_preview.png' alt='Learnk8s preview' />,
+} as const
+
+export const Blog = {
+  id: 'blog',
+  url: '/blog',
+  title: 'Blog ♦︎ Learnk8s',
+  description: 'The fastest way to become an expert in deploying applications at scale with Kubernetes.',
+}
+
+export function Register(store: Store<State, Actions>) {
+  store.dispatch(Action.registerPage(Blog))
+  store.dispatch(
+    Action.registerOpenGraph({
+      id: 'og-blog',
+      pageId: Blog.id,
+      image: <img src='assets/open_graph_preview.png' alt='Learnk8s preview' />,
+      title: 'blog',
+      description: 'The fastest way to become an expert in deploying applications at scale with Kubernetes.',
+    }),
+  )
+}
+
+export function Mount({ store }: { store: Store<State, Actions> }) {
+  const state = store.getState()
+  defaultAssetsPipeline({
+    jsx: renderPage(state),
+    isOptimisedBuild: getConfig(state).isProduction,
+    siteUrl: `${getConfig(state).protocol}://${getConfig(state).hostname}`,
+    url: Blog.url,
+    outputFolder: getConfig(state).outputFolder,
+  })
+}
+
+function renderPage(state: State) {
+  const page = getPages(state).find(it => it.id === Blog.id)!
+  const openGraph = getOpenGraph(state).find(it => it.pageId === Blog.id)
+  const currentAbsoluteUrl = `${state.config.protocol}://${join(state.config.hostname, page.url)}`
+  const blogPosts = getBlogPosts(state).sort(comparePublishedDate)
+  const allPages = state.website.pages

```

Comment:
You have already `allPages` on line 52

Reply:
But line 52 is only the blog page. `allPages` needed for all blog posts title and description.

Reply:
You're right. `getBlogPosts` then

Reply:
oh, I am Sorry, I am telling it wrong, `allPages` needed for all blog posts url, title and description are get from `getBlogPosts`.

Reply:
I think this commit `1282369` will make it better.